### PR TITLE
make compatible with python3

### PR DESCRIPTION
**Pull request by kkiesling**
(creating PR to to be able to run on CI, changes likely still needed)

This work is attempting to make all things compatible with python3. Changes made here need to be checked that they are still compatible with python2.

There are known python3 errors in the following files but I am unsure how to fix them:
`test_alara.py` - rounding error in one of the files generated
`test_mcnp.py` - rounding error

This PR addresses #49 .

### CODE REVIEW

**Comment by kkiesling** [comment]
CI needs to also be updated to test with python 3 and pymoab.

**Comment by gonuke** [comment]
I’m a little concerned that commit ae30824fdbe834f08a9bdbe7a0f36e078f0f1b5b was a merge commit instead of a rebase, for when it comes time to combine this with the main PR?

**Comment by gonuke** [comment]
I also think we need a tutorial on utf-8 and encodings

**Comment by kkiesling** [comment]
I agree on the utf-8 tutorial.. I am not on board even with some of my own changes here. I just am not sure the better way to do some of this.

File: tests/test_mesh.py [hunk 1]
```diff
@@ -5,7 +5,19 @@
...
+try:
+    from itertools import izip as zip
+except:
+    pass
+try:
+    from itertools import izip_longest as zip_longest
+except:
+    from itertools import zip_longest
+
+from itertools import zip_longest
```
Comment (kkiesling) [inline tests/test_mesh.py:17]: note to self: this line should be removed

File: tests/test_partisn.py [hunk 2]
```diff
@@ -397,7 +397,11 @@ def write_partisn_input_1D():
     ngroup = 5
 
     # expected output file
-    file_expected = THIS_DIR + '/files_test_partisn/partisn_1D_expected.inp'
+    import platform
+    if platform.python_version().split('.')[0] == '2':
```
Comment (kkiesling) [inline tests/test_partisn.py:401]: all these expected files that have a python 2 and python 3 version may not actually be necessary in the end. I need to check if the changes I made for python3 do the same thing in python 2 before I delete the original files (which are the python 2 version). Thanks to dictionaries no longer being ordered, things were being printed in different orders.

File: tests/test_mesh.py [hunk 3]
```diff
@@ -5,7 +5,19 @@
 import shutil
 import warnings
 import itertools
+try:
+    from itertools import izip as zip
+except:
+    pass
```
Comment (bam241) [inline tests/test_mesh.py:11]: is `izip` still required ?

**Comment by bam241** [comment]
I have issued a [PR](https://github.com/kkiesling/pyne/pull/2) against @kkiesling branch that should fix all the python2 problem raised by this... while keeping the tests working on `python3`

**Comment by kkiesling** [comment]
This PR being deferred until after PyMOAB is in PyNE. Will follow up with smaller PRs into pyne/pyne.

**Pull request closed without merging**